pkg/riff/commands/application_create.go

```diff
@@ -142,6 +146,22 @@ func (opts *ApplicationCreateOptions) Exec(ctx context.Context, c *cli.Config) e
 		return err
 	}
 	c.Successf("Created application %q\n", application.Name)
+	if opts.Tail {
```

Comment:
This block is repeated four times. It would be worth factoring out the common logic to avoid quadruple maintenance. If that was done, it may even be worth writing a unit test of the common logic to avoid having to repeat certain test code. This may be something for a future PR though.

Reply:
> This block is repeated four times. It would be worth factoring out the common logic to avoid quadruple maintenance.

The behavior waiting for a resource to become ready is already separate from the behavior that grabs the logs. I'd rather not conflate those two concerns.

> it may even be worth writing a unit test of the common logic to avoid having to repeat certain test code.

Test coverage for commands is quite high, but coverage for utility code is quite low and needs to be improved

> This may be something for a future PR though.

agreed